src/argilla/client/api.py

```diff
@@ -295,20 +314,30 @@ def load(
         >>> dataset_batch_2 = rg.load(name="example-dataset", limit=1000, id_from=dataset_batch_1[-1].id)
 
     """
-    return ArgillaSingleton.get().load(
-        name=name,
-        workspace=workspace,
-        query=query,
-        vector=vector,
-        ids=ids,
-        limit=limit,
-        sort=sort,
-        id_from=id_from,
-        batch_size=batch_size,
-        include_metrics=include_metrics,
-        include_vectors=include_vectors,
-        as_pandas=as_pandas,
-    )
+    argilla = ArgillaSingleton.get()
+    try:
+        return argilla.load(
+            name=name,
+            workspace=workspace,
+            query=query,
+            vector=vector,
+            ids=ids,
+            limit=limit,
+            sort=sort,
+            id_from=id_from,
+            batch_size=batch_size,
+            include_metrics=include_metrics,
+            include_vectors=include_vectors,
+            as_pandas=as_pandas,
+        )
+    except errors.NotFoundApiError as e:
+        workspace = workspace or argilla.get_workspace()
+        if _check_if_feedback_dataset_exists(client=argilla, name=name, workspace=workspace):
+            raise ValueError(
+                f"The dataset '{name}' exists but it is a `FeedbackDataset`. Use `rg.FeedbackDataset.from_argilla`"
+                " instead to load it."
+            )
```

Comment:
I'm not sure if a warning would be better than an exception only for this.

Reply:
I think it is better to stop the execution of the program as soon as possible. Maybe some users decide to suppress the `argilla` log messages and do not see the warning message, and get a `None` instead and make them even more confused.